lib/index.js

```diff
@@ -405,14 +405,13 @@ FSTree.prototype.match = function(globs) {
 };
 
 FSTree.prototype.changes = function() {
-  var patches;
+  let patches;
+  this._ensureEntriesPopulated();
   if(this.srcTree ) {
-    this._ensureEntriesPopulated();
     var prevTree = new FSTree.fromEntries(this.prevEntries);
     patches = prevTree.calculatePatch(this);
     this.prevEntries = [];
-    this.entries.forEach(e => this.prevEntries.push(e));
-
+    this.prevEntries = this.entries.slice(0, this.entries.length);
```

Comment:
I think this can be simplified to:

    this.prevEntries = this.entries.slice();